imagebuildah/stage_executor.go

```diff
@@ -966,19 +967,20 @@ func (s *StageExecutor) prepare(ctx context.Context, from string, initializeIBCo
 		}
 	}
 
-	builderSystemContext := s.executor.systemContext
-	// get platform string from stage
-	if stage.Builder.Platform != "" {
-		os, arch, variant, err := parse.Platform(stage.Builder.Platform)
+	// In a multi-stage build where `FROM --platform=<>` was used then we must
+	// reset context for new stages so that new stages don't inherit unexpected
+	// `--platform` from prior stages.
+	if stage.Builder.Platform != "" || (len(s.stages) > 1 && ((s.systemContext.ArchitectureChoice == "" && s.systemContext.VariantChoice == "" && s.systemContext.OSChoice == "") || (s.executor.skipUnusedStages == types.OptionalBoolFalse))) {
```

Comment:
In the current revision, `s.executor.skipUnusedStages` doesn't appear to matter and can likely be dropped, as the current revision ensures that OSChoice/ArchitectureChoice/VariantChoice is explicitly specified for every stage of a multi-stage build (earlier revisions only did so for the second and later stages), either to values given in the `--platform` flag for the FROM instruction, values passed in through our API, or the current system's defaults.

Reply:
I removed this.